### ISSUE TITLE
["Request"] Consider creating a BOM

### ISSUE THREAD

**Issue by ps-feng** [arrow-kt/arrow#2129]
### What version are you currently using?
0.10.5

### What would you like to see?
I wonder if you've considered providing a BOM so that Arrow's version numbers only need to appear once in the dependencies. Usage in Gradle is then simplified a bit:
```kotlin
dependencies {
    implementation(platform("io.arrow-kt:arrow-stack:$arrowVersion"))

    // versions no longer required here
    implementation("io.arrow-kt:arrow-core")
    implementation("io.arrow-kt:arrow-fx")
    implementation("io.arrow-kt:arrow-mtl")
    implementation("io.arrow-kt:arrow-syntax")
}
```
It also opens the possibility for diverging versions in modules without increasing the dependency burden on the user's side, even though I guess that's generally not desirable.

**Comment by pakoito** [comment]
@rachelcarmena I believe we had this at some point, I don't know where it ended up with the repo split.

**Comment by rachelcarmena** [comment]
Thank you so much @ps-feng :raised_hands: 

> @rachelcarmena I believe we had this at some point, I don't know where it ended up with the repo split.

@pakoito , please, could you provide me more details? I didn't find anything about it:

* Just JAR artifacts with the correspondent POM file in [OSS](https://oss.jfrog.org/artifactory/oss-snapshot-local/io/arrow-kt/). 
* Nothing about publishing a BOM file in `arrow` repository for old versions.

Don't worry if nothing exists. Let's create it!

**Comment by rachelcarmena** [comment]
It will be solved via #2131

**Comment by pakoito** [comment]
Maybe it was `arrow-shadow` or `arrow-bom`, I don't remember.

**Comment by rachelcarmena** [comment]
> Maybe it was `arrow-shadow` or `arrow-bom`, I don't remember.

I didn't find any BOM file in OSS repository. `arrow-bom` doesn't exist. `arrow-shadow` is a shadowed JAR which is not published since 0.9x.

Don't worry! `arrow-stack` already exists! :raised_hands: https://oss.jfrog.org/artifactory/oss-snapshot-local/io/arrow-kt/arrow-stack/ and it will be published in Maven Central for the next release.

I created an example of use: https://github.com/arrow-kt/arrow-examples/pull/4

Thanks again @ps-feng! Closing the issue. Please, reopen it in case of a missing feature :raised_hands:
